File: src/test/java/ru/chernov/urlshortener/helper/HttpHelper.java
package ru.chernov.urlshortener.helper;

import org.springframework.http.MediaType;
import org.springframework.test.web.servlet.request.MockHttpServletRequestBuilder;

import static org.springframework.restdocs.mockmvc.RestDocumentationRequestBuilders.delete;
import static org.springframework.restdocs.mockmvc.RestDocumentationRequestBuilders.get;
import static org.springframework.restdocs.mockmvc.RestDocumentationRequestBuilders.post;
import static org.springframework.restdocs.mockmvc.RestDocumentationRequestBuilders.put;


public class HttpHelper {

    public static MockHttpServletRequestBuilder getJson(String url, Object... urlVariables) {
        return get(url, urlVariables).contentType(MediaType.APPLICATION_JSON);
    }


    public static MockHttpServletRequestBuilder postJson(String url, Object... urlVariables) {
        return post(url, urlVariables).contentType(MediaType.APPLICATION_JSON);
    }


    public static MockHttpServletRequestBuilder putJson(String url, Object... urlVariables) {
        return put(url, urlVariables).contentType(MediaType.APPLICATION_JSON);
    }


    public static MockHttpServletRequestBuilder deleteJson(String url, Object... urlVariables) {
        return delete(url, urlVariables).contentType(MediaType.APPLICATION_JSON);
    }

}

File: src/test/kotlin/ru/chernov/urlshortener/helper/HttpHelper.kt
package ru.chernov.urlshortener.helper

import org.springframework.http.MediaType
import org.springframework.restdocs.mockmvc.RestDocumentationRequestBuilders.*
import org.springframework.test.web.servlet.request.MockHttpServletRequestBuilder


fun getJson(url: String?, vararg urlVariables: Any?): MockHttpServletRequestBuilder {
    return get(url, urlVariables).contentType(MediaType.APPLICATION_JSON)
}


fun postJson(url: String?, vararg urlVariables: Any?): MockHttpServletRequestBuilder {
    return post(url, urlVariables).contentType(MediaType.APPLICATION_JSON)
}


fun putJson(url: String?, vararg urlVariables: Any?): MockHttpServletRequestBuilder {
    return put(url, urlVariables).contentType(MediaType.APPLICATION_JSON)
}


fun deleteJson(url: String?, vararg urlVariables: Any?): MockHttpServletRequestBuilder {
    return delete(url, urlVariables).contentType(MediaType.APPLICATION_JSON)
}